c_api/index_io_c.cpp

```diff
@@ -83,3 +84,13 @@ int faiss_read_index_binary_fname(
     }
     CATCH_AND_HANDLE
 }
+
+int faiss_read_VectorTransform_fname(
```

Comment:
Could we make the change to the function name to all lower cases like faiss_read_vector_transform()

Reply:
This is merely following the convention already applied for the rest of the C API. As the underlying object being created is of type `VectorTransform`, the function signature retains that name to let users understand what it maps to in the C++ API.
IMO it would be too disruptive to start changing this convention now.

I also don't think we have any unit tests for the C API at the moment, so we'd need to kickstart that.